apps/meteor/app/federation-v2/server/infrastructure/rocket-chat/adapters/Settings.ts

```diff
@@ -202,8 +234,30 @@ export class RocketChatSettingsAdapter {
 					type: 'code',
 					i18nLabel: 'Federation_Matrix_registration_file',
 					i18nDescription: 'Federation_Matrix_registration_file_desc',
+					alert: 'Federation_Matrix_registration_file_Alert',
 				});
 			});
 		});
 	}
+
+	private getRegistrationFileFromHomeserver(): Record<string, any> | undefined {
+		try {
+			const registrationYaml = Assets.getText('federation/registration.yaml');
```

Comment:
@RocketChat/chat-engine The purpose here is to read a file from a place in the File System in order to enable/disable the setting. We are doing that to force the user to strictly enable it using the Matrix homeserver config file.

I'm following @alansikora suggestion here to read the file from the `private` folder and use the `Assets.getText()` method. I'm not sure if this is the best way to go, appreciate any suggestion. Thanks.

Reply:
this would mean people would have to build rocket.chat from source after creating this file, is that what you want to be done?

Reply:
For sure, it's not. So, what is the proper way to do that?

The final desire here is to be able to enable/disable a setting only and only through this file (it's the same file used to configure the Matrix Bridge).

Is there any other (and suggested) folder to place this file? Also, should we use native NodeJS `fs` instead of `Assets.getText()`?

Reply:
@sampaiodiego I've changed it to use `fs` reading from a folder in the meteor root project.